### PR TITLE
Update to use latest 3.0.x stable

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

File: src/test/java/io/crate/integrationtests/FromURLPropertyClusterTest.java [hunk 1]
```diff
@@ -49,6 +50,7 @@
         .fromSysProperties()
         .clusterName(CLUSTER_NAME)
         .numberOfNodes(2)
+        .settings(ImmutableMap.of("node.max_local_storage_nodes", (Object) 2))
```
Comment (seut) [inline src/test/java/io/crate/integrationtests/FromURLPropertyClusterTest.java]: using `ImmutableMap.<String, Object>of()` would avoid the cast

File: src/test/java/io/crate/integrationtests/FromVersionPropertyClusterTest.java [hunk 2]
```diff
@@ -49,6 +50,7 @@
         .fromSysProperties()
         .clusterName(CLUSTER_NAME)
         .numberOfNodes(2)
+        .settings(ImmutableMap.of("node.max_local_storage_nodes", (Object) 2))
```
Comment (seut) [inline src/test/java/io/crate/integrationtests/FromVersionPropertyClusterTest.java]: also here

File: src/test/java/io/crate/integrationtests/FromURLPropertyClusterTest.java (hunk 1 above)
Comment (seut) [reply src/test/java/io/crate/integrationtests/FromURLPropertyClusterTest.java]: maybe it makes sense to set this automatically based on `.numberOfNodes()`?

File: src/test/java/io/crate/testing/ClusterTest.java [hunk 3]
```diff
@@ -39,7 +40,7 @@
 public class ClusterTest extends BaseTest {
 
     private static final String CLUSTER_NAME = "cluster";
-    private static final String VERSION = "0.57.1";
+    private static final String VERSION = "3.0.3";
```
Comment (seut) [inline src/test/java/io/crate/testing/ClusterTest.java]: maybe it makes sense to make this public and use it everywhere? (or move it to a kind of config class)

File: src/test/java/io/crate/testing/ClusterTest.java [hunk 4]
```diff
@@ -49,14 +50,12 @@ public void testClusterBuilder() throws Throwable {
         CrateTestCluster cluster = CrateTestCluster.fromVersion(VERSION)
             .clusterName(CLUSTER_NAME)
             .numberOfNodes(2)
-            .settings(new HashMap<String, Object>() {{
-                put("stats.enabled", true);
-            }})
+            .settings(ImmutableMap.of("stats.enabled", (Object) true,
```
Comment (seut) [inline src/test/java/io/crate/testing/ClusterTest.java]: also here, using generics would avoid the cast

**Comment by matriv** [comment]
@seut Thx for the comments! Fixup pushed.

File: src/main/java/io/crate/testing/CrateTestCluster.java [hunk 5]
```diff
@@ -148,6 +152,10 @@ public Builder numberOfNodes(int numberOfNodes) {
                 throw new IllegalArgumentException(String.format("invalid number of nodes: %d", numberOfNodes));
             }
             this.numberOfNodes = numberOfNodes;
+            if (settings.isEmpty()) {
+                settings = new HashMap<>();
+            }
+            settings.put("node.max_local_storage_nodes", numberOfNodes);
```
Comment (seut) [inline src/main/java/io/crate/testing/CrateTestCluster.java]: is this safe for CrateDB < 3.0 ?

Comment (matriv) [reply src/main/java/io/crate/testing/CrateTestCluster.java]: Yes, it's safe, tested with versions 1.0.x up to 3.0.x

**Pull request merged**